### PR TITLE
Tests: now autoload tests from /worlds/*/test

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
now autoload tests from /worlds/*/test

## How was this tested?
:)

## If this makes graphical changes, please attach screenshots.
image

### CODE REVIEW

**Comment by Berserker66** [comment]
Not really sure why pytest dies on windows. I tested on windows, but I don't use pytest locally.

File: test/programs/TestGenerate.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class TestGenerateMain(unittest.TestCase):
     """This tests Generate.py (ArchipelagoGenerate.exe) main"""
 
     generate_dir = Path(Generate.__file__).parent
-    run_dir = generate_dir / 'test'  # reproducible cwd that's neither __file__ nor Generate.__file__
+    run_dir = generate_dir  # reproducible cwd that's neither __file__ nor Generate.__file__
```
Comment (black-sliver) [inline test/programs/TestGenerate.py:19]: why? this puts run_dir == source dir, which is specifically not what we want.

File: test/programs/TestGenerate.py [hunk 2]
```diff
@@ -30,12 +30,25 @@ def assertOutput(self, output_dir: str):
...
+        self.original_cached_path = Utils.local_path.cached_path
         Utils.local_path.cached_path = str(self.generate_dir)
         os.chdir(self.run_dir)
         self.output_tempdir = TemporaryDirectory(prefix='AP_out_')
 
     def tearDown(self):
         self.output_tempdir.cleanup()
+        os.chdir(self.original_cwd)
+        sys.argv = self.original_argv
+        Utils.local_path.cached_path = self.original_cached_path
```
Comment (black-sliver) [inline test/programs/TestGenerate.py:44]: tests should run isolated, so this does nothing

File: test/programs/TestGenerate.py [hunk 3]
```diff
@@ -58,7 +71,7 @@ def test_generate_relative(self):
     def test_generate_yaml(self):
         # override host.yaml
         defaults = Utils.get_options()["generator"]
-        defaults["player_files_path"] = str(self.yaml_input_dir)
+        defaults["player_files_path"] = os.path.abspath(self.yaml_input_dir)
```
Comment (black-sliver) [inline test/programs/TestGenerate.py:74]: any specific reason to not just use the relative path?

**Comment by Berserker66** [comment]
While github won't allow me, I do approve of Black Sliver's changes for the purposes of review.

**Review by black-sliver — APPROVED** [review]

**Pull request merged**